composer.json

```diff
@@ -22,7 +22,8 @@
     "php": ">=5.6.0",
     "ext-curl": "*",
     "ext-gmp": "*",
```

Comment:
@brianrlewis: One quick question, if the gmp is optional, will this line enforces the gmp to be installed, and if it is missing will it throw an error?

Reply:
@FYJen Yes, when composer is installed it will check the host system for that extension and will raise an error if it is missing. Since we are encouraging users to have that extension installed whenever possible it probably makes sense to keep it listed as a requirement in `composer.json`.

Users can always do a composer install with the `--ignore-platform-reqs` flag to make composer ignore that requirement if they so choose.